src/main/java/com/aws/greengrass/util/platforms/windows/WindowsRunasProcess.java

```diff
@@ -0,0 +1,450 @@
+import com.sun.jna.platform.win32.Win32Exception;
+import com.sun.jna.platform.win32.WinBase;
+import com.sun.jna.platform.win32.WinDef;
+import com.sun.jna.platform.win32.WinError;
+import com.sun.jna.platform.win32.WinNT;
+import com.sun.jna.ptr.IntByReference;
+import lombok.Getter;
+import vendored.com.microsoft.alm.storage.windows.internal.WindowsCredUtils;
+
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manage a Windows process running as a specific user.
+ */
+public class WindowsRunasProcess extends Process {
+    private static final int PROCESS_CREATION_FLAGS = WinBase.CREATE_UNICODE_ENVIRONMENT  // use unicode
+            | WinBase.CREATE_NO_WINDOW;  // don't create a window on desktop
+    private static final char NULL_CHAR = '\0';
+    public static final String SYSTEM_ROOT = "SystemRoot";
+    public static final String SERVICE_GROUP_SID = "S-1-5-6";
+    public static final int EXIT_CODE_TERMINATED = 130;
+
+    private final String domain;
+    private final String username;
+    private final Map<String, String> additionalEnv = new HashMap<>();
+    private String currentDirectory;
+
+    @Getter
+    private int pid = 0;
+    private WinBase.PROCESS_INFORMATION procInfo;
+    private boolean exited = false;
+    private int exitCode = -1;
+
+    private InputStream stdout;
+    private InputStream stderr;
+    private OutputStream stdin;
+
+    private WinNT.HANDLEByReference inPipeReadHandle;
+    private WinNT.HANDLEByReference inPipeWriteHandle;
+    private WinNT.HANDLEByReference outPipeReadHandle;
+    private WinNT.HANDLEByReference outPipeWriteHandle;
+    private WinNT.HANDLEByReference errPipeReadHandle;
+    private WinNT.HANDLEByReference errPipeWriteHandle;
+
+    /**
+     * Setup to run a Windows process as a specific user.
+     *
+     * @param domain   name of domain that contains the user account
+     * @param username name of the account
+     */
+    public WindowsRunasProcess(String domain, String username) {
+        super();
+        this.domain = domain;
+        this.username = username;
+    }
+
+    /**
+     * Set env vars in addition to the user's own existing env vars.
+     *
+     * @param envs a map of env vars to set
+     */
+    public void setEnv(Map<String, String> envs) {
+        additionalEnv.putAll(envs);
+    }
+
+    public synchronized void setCurrentDirectory(String dir) {
+        currentDirectory = dir;
+    }
+
+    /**
+     * Start the process as given user.
+     *
+     * @param command command line to run
+     * @throws IOException if failed to run subprocess
+     */
+    public synchronized void start(String command) throws IOException {
```

Comment:
needs more debug logs everywhere (especially after deciding which API method to use)

maybe add trace logs for each JNA call too?